LearnNet_CatalogService/Controllers/CategoryController.cs

```diff
@@ -0,0 +1,189 @@
+﻿using FluentValidation;
+using LearnNet_CatalogService.Core.DTO;
+using LearnNet_CatalogService.Core.Interfaces;
+using LearnNet_CatalogService.HATEOAS;
+using LearnNet_CatalogService.Models.Category;
+using Microsoft.AspNetCore.Mvc;
+
+// For more information on enabling Web API for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
+
+namespace LearnNet_CatalogService.Controllers
+{
+    [Route("api/category")]
+    [ApiController]
+    public class CategoryController : ControllerBase
+    {
+        private readonly ICategoryService _categoryService;
+        private readonly ILogger<CategoryController> _logger;
+        private readonly LinkGenerator _linkGenerator;
+        private readonly IValidator<CategoryDTO> _validator;
+
+        public CategoryController(ICategoryService categoryService,
+                                  ILogger<CategoryController> logger,
+                                  LinkGenerator linkGenerator,
+                                  IValidator<CategoryDTO> validator)
+        {
+            _categoryService = categoryService;
+            _logger = logger;
+            _linkGenerator = linkGenerator;
+            _validator = validator;
+        }
+
+        // GET: api/<CategoryController>
+        [HttpGet]
```

Comment:
Please mark each action with appropriate [ProducesResponseType] attributes
https://medium.com/@mohammed0hamdan/using-producesresponsetype-to-write-a-better-web-api-actions-in-net-core-18e080c9bf00